docs/content/en/docs/Reference/configuration.md

```diff
@@ -434,3 +469,538 @@ vpn:
+```
+
+### Filtering stages with if statement
+
+Stages can be skipped based on if statements:
+
+
+```yaml
+#cloud-config
+
+stages:
+  foo:
+  - name: "echo"
+    commands:
+    - echo hello
+    if: "cat /proc/cmdline | grep debug"
+
+name: "Test yip!"
+```
+
+The expression inside the if will be evaluated in bash and, if specified, the stage gets executed only if the condition returns successfully (exit 0).
+
+
+### `name`
+
+A description of the stage step. Used only when printing output to console.
+
+### `commands`
+
+A list of arbitrary commands to run after file writes and directory creation.
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup something"
+       commands:
+         - echo 1 > /bar
+```
+
+### `files`
+
+A list of files to write to disk.
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - files:
+        - path: /tmp/bar
+          content: |
+                    #!/bin/sh
+                    echo "test"
+          permissions: 0777
+          owner: 1000
+          group: 100
+```
+
+### `directories`
+
+A list of directories to be created on disk. Runs before `files`.
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup folders"
+       directories:
+       - path: "/etc/foo"
+         permissions: 0600
+         owner: 0
+         group: 0
+```
+
+### `dns`
+
+A way to configure the `/etc/resolv.conf` file.
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup dns"
+       dns:
+         nameservers:
+         - 8.8.8.8
+         - 1.1.1.1
+         search:
+         - foo.bar
+         options:
+         - ..
+         path: "/etc/resolv.conf.bak"
+```
+### `hostname`
+
+A string representing the machine hostname. It sets it in the running system, updates `/etc/hostname` and adds the new hostname to `/etc/hosts`.
+Templates can be used to allow dynamic configuration. For example in mass-install scenario it could be needed (and easier) to specify hostnames for multiple machines from a single cloud-init config file.
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup hostname"
+       hostname: "node-{{ trunc 4 .MachineID }}"
+```
+### `sysctl`
+
+Kernel configuration. It sets `/proc/sys/<key>` accordingly, similarly to `sysctl`.
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup exception trace"
+       systctl:
+         debug.exception-trace: "0"
+```
+
+### `authorized_keys`
+
+A list of SSH authorized keys that should be added for each user.
+SSH keys can be obtained from GitHub user accounts by using the format github:${USERNAME},  similarly for Gitlab with gitlab:${USERNAME}.
+
+```yaml
+#cloud-config
+
+stages:
+   boot:
+     - name: "Setup exception trace"
+       authorized_keys:
+         mudler:
+         - github:mudler
+         - ssh-rsa: ...
+```
+
+### `node`
+
+If defined, the node hostname where this stage has to run, otherwise it skips the execution. The node can be also a regexp in the Golang format.
```

Comment:
```suggestion
If defined, the node hostname where this stage has to run, otherwise it skips the execution. The node can also be a regexp in the Golang format.
```